lib/services/local/autoupdate.go

```diff
@@ -91,11 +92,24 @@ func NewAutoUpdateService(b backend.Backend) (*AutoUpdateService, error) {
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
+	report, err := generic.NewServiceWrapper(
+		generic.ServiceConfig[*autoupdate.AutoUpdateAgentReport]{
+			Backend:       b,
+			ResourceKind:  types.KindAutoUpdateAgentRollout,
+			BackendPrefix: backend.NewKey(autoUpdateAgentRolloutPrefix),
+			MarshalFunc:   services.MarshalProtoResource[*autoupdate.AutoUpdateAgentReport],
+			UnmarshalFunc: services.UnmarshalProtoResource[*autoupdate.AutoUpdateAgentReport],
+			ValidateFunc:  update.ValidateAutoUpdateAgentReport,
```

Comment:
This (the traditional way) will prevent getting and listing if an item turns out to be broken, it would be a better pattern to allow reading anything that successfully unmarshals and validate before writing and before use, so you get to actually list and see the problematic thing instead of having to guess at what to delete.

Reply:
In this case, why don't we change the generic adapter to run the validate func only before writes?

Reply:
That would also imply that every reader should handle the possibility that an item doesn't pass CheckAndSetDefaults, which means changing every (existing) callsite.

Reply:
Gotcha, we should update RFD 153 and deprecate the ValidateFunc field so people stop using it in new code.